docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch

```diff
@@ -0,0 +1,145 @@
+From ddf20fca1e5d3f5669909e825b6b20332887d006 Mon Sep 17 00:00:00 2001
+From: Quentin Boyer <quentin+dev@familleboyer.net>
+Date: Wed, 11 Sep 2024 00:05:47 +0200
+Subject: [PATCH] nixos-render-docs: Support gfm-style admonitions
+
+---
+ nixos_render_docs/md.py  | 35 +++++++++++++++++++++++++++++++++++
+ tests/sample_md.py       |  3 +++
+ tests/test_asciidoc.py   |  6 ++++++
+ tests/test_commonmark.py |  3 +++
+ tests/test_html.py       |  4 ++++
+ tests/test_manpage.py    |  6 ++++++
+ 6 files changed, 57 insertions(+)
+
+diff --git a/nixos_render_docs/md.py b/nixos_render_docs/md.py
+index 894daf9ca9c7..cde874bc5637 100644
+--- a/nixos_render_docs/md.py
++++ b/nixos_render_docs/md.py
+@@ -341,6 +341,40 @@ def _parse_blockattrs(info: str) -> Optional[tuple[AttrBlockKind, Optional[str],
+         return ('figure', id, classes)
+     return None
+ 
++_ALERT_PATTERN = re.compile(r"^\[\!(TIP|NOTE|IMPORTANT|WARNING|CAUTION)\]", re.IGNORECASE)
++
++def _gfm_alert(md: markdown_it.MarkdownIt):
++    def gfm_alert_to_adm(state: markdown_it.rules_core.StateCore):
++        tokens = state.tokens
++        for (i, token) in enumerate(tokens):
++            if token.type == "blockquote_open":
++                open = tokens[i]
++                start_index = i
++                while tokens[i].type != "blockquote_close" and i < len(tokens):
++                    i += 1
++                close = tokens[i]
++                end_index = i
++
++                first_content = next(
++                    (t for t in tokens[start_index:end_index + 1] if t.type == "inline"),
++                    None,
++                )
++                if first_content is None:
++                    continue
++
++                if (m := _ALERT_PATTERN.match(first_content.content)) is None:
++                    continue
++
++                first_content.content = first_content.content[len(m[0]):].lstrip()
```

Comment:
```suggestion
+                # Remove "> [!TIP]\n" from the token
+                first_content.content = first_content.content[m.endpos:].lstrip()
```

- Let's use `Match.endpos` to get the substring index
- Let's add a comment so this makes sense in 12 months time :upside_down_face: 
- Do we really need `lstrip()`? I think it would be better to match whitespace in the regex.

Reply:
`endpos` is not the same as `len(m[0])`:

> The value of endpos which was passed to the [search()](https://docs.python.org/3/library/re.html#re.Pattern.search) or > [match()](https://docs.python.org/3/library/re.html#re.Pattern.match) method of a [regex object](https://docs.python.org/3/library/re.html#re-objects). This is the index into the string beyond which the RE engine will not go.

In pratice here it is the length of the string

Reply:
Ah, I guess we want [`m.end(0)`](https://docs.python.org/3/library/re.html#re.Match.end) then. Sorry, just doing my best to read the docs and usually reading the wrong part :sweat_smile: